variables.tf

```
@@ -1,15 +1,57 @@
-
+variable "bucket_name" {
+  default     = null
+  description = "Name to apply to bucket (use `bucket_name` or `bucket_suffix`)"
+  type        = string
+}
 
 variable "bucket_suffix" {
-  type = string
+  default     = "default"
+  description = "Suffix to apply to the bucket (use `bucket_name` or `bucket_suffix`). When using `bucket_suffix`, the bucket name will be `[ACCOUNT_ID]-[REGION]-s3logging-[BUCKET_SUFFIX]."
+  type        = string
+}
+
+variable "lifecycle_rules" {
+  default = [{
+    id                            = "expire"
+    enabled                       = true
+    prefix                        = null
+    expiration                    = 2147483647
```

Because of the way you're setting the default in the code, and the way I think you _should_ set it, I think a null default here is better.

```suggestion
    expiration                    = null
```

---

You can't set this to a null, the block requires it, and AWS doesn't except a value that means "don't expire". Their recommendation is this admittedly absurd value.

---

@cdaniluk but your code includes that as a default already, and I had suggested removing the entire block if this is undefined by using a `try()` (if possible)

https://github.com/rhythmictech/terraform-aws-s3logging-bucket/blob/815b74ca3967330d7588d0442ecbacdfb5b289b2/main.tf#L33-L35

---

`try()` only works as a function, it cannot make a block itself null. If the block is present, the expiration must be set to a non-negative integer per AWS support:

> Kindly note that the maximum expiration value (in days) that you can set in a lifecycle policy is 2147483647. On the other hand, we cannot specify 0 with expiration. Expire after days must be a positive integer.

I agree with your logic, it just isn't supported by what the provider and `try()` can do

---

I want to find a way to do it, but not worth blocking this PR on it.